centos/nap/entrypoint.sh

```diff
@@ -0,0 +1,112 @@
+#!/bin/sh
+#
+# This script launches nginx and the NGINX Controller Agent.
+#
+# If several instances use the same imagename, the metrics will
+# be aggregated into a single object in Controller. Otherwise NGINX Controller
+# will create separate objects for monitoring (an object per instance).
+
+# Variables
+agent_conf_file="/etc/controller-agent/agent.conf"
+agent_log_file="/var/log/nginx-controller/agent.log"
+nginx_status_conf="/etc/nginx/conf.d/stub_status.conf"
+api_key=""
+instance_name="$(hostname -f)"
+controller_url=""
+location=""
+
+handle_term()
+{
+    echo "received TERM signal"
+    echo "stopping controller-agent ..."
+    kill -TERM "${agent_pid}" 2>/dev/null
+    echo "stopping nginx ..."
+    kill -TERM "${nginx_pid}" 2>/dev/null
+}
+
+trap 'handle_term' TERM
+
+# Launch NAP
+/bin/su -s /bin/bash -c '/opt/app_protect/bin/bd_agent &' nginx
```

Comment:
You just run this in the background and do not handle any kind of graceful shutdown. Do that as in the case of agent or Nginx(same in all entrypoint versions)